app/services/form_element_builder.rb

```diff
@@ -23,15 +23,25 @@ def element
   end
 
   def params
-    if @params[:choices].respond_to?(:map)
-      @params[:choices].map! do |choice|
-        begin
-          JSON.parse(choice)
-        rescue JSON::ParserError
-          choice
-        end
+    @params[:choices] = format_many_choices if @params[:many_choices].present?
+    @params[:choices] = format_choices
```

Comment:
because of the assignment I'd rather you passed `params[:choices]` to `format_many_choices` and `format_choices`. I think it makes the code easier to reason with.